File: server/models/address.js
/**
 * 收货地址模型 → addresses 表
 */
const { DataTypes } = require('sequelize')
const sequelize = require('../config/database')

const Address = sequelize.define('Address', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    comment: '所属用户 id',
  },
  receiver: { type: DataTypes.STRING(50), allowNull: false, comment: '收货人姓名' },
  phone: { type: DataTypes.STRING(20), allowNull: false, comment: '收货人电话' },
  province: { type: DataTypes.STRING(30), defaultValue: '', comment: '省' },
  city: { type: DataTypes.STRING(30), defaultValue: '', comment: '市' },
  district: { type: DataTypes.STRING(30), defaultValue: '', comment: '区县' },
  detail: { type: DataTypes.STRING(200), allowNull: false, comment: '详细地址' },
  is_default: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    comment: '是否默认地址（同一用户仅一条为 true）',
  },
}, {
  tableName: 'addresses',
})

module.exports = Address

File: springboot/src/main/java/com/freshmall/entity/Address.java
package com.freshmall.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 收货地址表
 */
@Data
@Entity
@Table(name = "addresses")
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "user_id", nullable = false)
    private Integer user_id;

    @Column(nullable = false, length = 50)
    private String receiver;

    @Column(nullable = false, length = 20)
    private String phone;

    @Column(length = 30)
    private String province = "";

    @Column(length = 30)
    private String city = "";

    @Column(length = 30)
    private String district = "";

    @Column(nullable = false, length = 200)
    private String detail;

    @Column(name = "is_default")
    private Boolean is_default = false;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
